core/shared/src/main/scala/scalaz/zio/internal/Scheduler.scala

```diff
@@ -0,0 +1,72 @@
+// Copyright (C) 2018 - 2019 John A. De Goes. All rights reserved.
+package scalaz.zio.internal
+
+import scala.concurrent.duration._
+import java.util.concurrent._
+import java.util.concurrent.atomic.AtomicInteger
+
+trait Scheduler {
+  import Scheduler.CancelToken
+
+  def schedule(executor: Executor, task: Runnable, duration: Duration): CancelToken
+
+  /**
+   * The number of tasks scheduled.
+   */
+  def size: Int
+
+  /**
+   * Initiates shutdown of the scheduler.
+   */
+  def shutdown(): Unit
+}
+
+object Scheduler {
+  type CancelToken = () => Boolean
```

Comment:
no mayInterruptIfRunning?

Reply:
Nope, no need, that's handled by `IO` directly.